### ISSUE TITLE
Experiment #006 - Repeat Experiment #005 with different data

### ISSUE THREAD

**Issue by AyronK** [AyronK/softmax-dnn-example#12]
I'd like to test #8 with decks of 4 cards and only one duplicate to not complicate the case.

## Assumptions
1. Full deck has 4 card slots
2. There is 10 total cards in the pool.
3. Cards are values of one in 10-length array:
   [1,0,0,0,0,0,0,0,0,0] - first card
   [1,0,1,0,1,0,0,0,0,0] - deck containing first, third and fifth card
4. Data is correlated:
```
                //Class2
                new double[] {2, 1, 0, 0, 1, 0, 0, 0, 0, 0},
                new double[] {1, 2, 0, 0, 1, 0, 0, 0, 0, 0},
                new double[] {2, 1, 0, 0, 0, 1, 0, 0, 0, 0},
                new double[] {1, 2, 0, 0, 0, 1, 0, 0, 0, 0},

                new double[] {2, 1, 0, 0, 1, 0, 0, 0, 0, 0},
                new double[] {2, 1, 0, 0, 1, 0, 0, 0, 0, 0},
                new double[] {2, 1, 0, 0, 0, 1, 0, 0, 0, 0},
                new double[] {2, 1, 0, 0, 0, 1, 0, 0, 0, 0},

                new double[] {1, 2, 0, 0, 1, 0, 0, 0, 0, 0},
                new double[] {1, 2, 0, 0, 1, 0, 0, 0, 0, 0},
                new double[] {1, 2, 0, 0, 0, 1, 0, 0, 0, 0},
                new double[] {1, 2, 0, 0, 0, 1, 0, 0, 0, 0},

                //Class1
                new double[] {0, 0, 2, 1, 0, 0, 0, 0, 1, 0},
                new double[] {0, 0, 1, 2, 0, 0, 0, 0, 1, 0},
                new double[] {0, 0, 2, 1, 0, 0, 0, 0, 0, 1},
                new double[] {0, 0, 1, 2, 0, 0, 0, 0, 0, 1},

                new double[] {0, 0, 2, 1, 0, 0, 0, 0, 1, 0},
                new double[] {0, 0, 2, 1, 0, 0, 0, 0, 1, 0},
                new double[] {0, 0, 2, 1, 0, 0, 0, 0, 0, 1},
                new double[] {0, 0, 2, 1, 0, 0, 0, 0, 0, 1},

                new double[] {0, 0, 1, 2, 0, 0, 0, 0, 1, 0},
                new double[] {0, 0, 1, 2, 0, 0, 0, 0, 1, 0},
                new double[] {0, 0, 1, 2, 0, 0, 0, 0, 0, 1},
                new double[] {0, 0, 1, 2, 0, 0, 0, 0, 0, 1},
```
Same rules for card usage.
I tried to create as much correlated data as possible, to test duplicates properly.

## Process
1. For each deck remove one "card" and take two as input and one as output (if there is a duplicate we only use this case once - this means we will have 3 inputs for one deck)
    - input:   [2,0,0,0,1,0,0,0,0,0] 
    - output: [0,0,1,0,0,0,0,0,0,0] 
    - **OR for duplicates**
    - input:   [1,0,1,0,1,0,0,0,0,0] 
    - output: [1,0,0,0,0,0,0,0,0,0] 
2. Shuffle the data of 72 total (_24 decks * 3 cards_) options
3. Take 24 options as learning set and 48 as testing set
4. Teach network
5. Test accuracy
6. Test how duplicates affect on the results

**Comment by AyronK** [comment]
## Result
Below the comparison between avarage of 100 tries accuracy for two different networks

### Activation Network
- 5 hidden neurons
- LearningRate = 0.07,
- Momentum = 0.3

**Learing set accuracy:** 89,92%
**Testing set accuracy:** 76,54%

### Deep Belief Neural Network
- 2 hidden layers, 5 neurons each
- Unsupervised
   * LearningRate = 0.07,
   * Momentum = 0.3
- Supervised
   * LearningRate = 0.07,
   * Momentum = 0.2

**Learing set accuracy:** 91,00%
**Testing set accuracy:** 73,94%

**Comment by AyronK** [comment]
## Conclusion

- Focusing on duplicates only gave much better results
- Indicating duplicate as value of 2 might be good enough.

Further tests on real data in #9 are required to verify - there is no need to be afraid of decks with duplicates to teach on.
